### ISSUE TITLE
Cloud SQL deployment is failing with unknown reason

### ISSUE THREAD

**Issue by DenisLaboureyras** [pulumi/pulumi-gcp#970]
### What happened?

Since yesterday, my pulumi stack is failing every time I try to deploy because of an unexpected update on the Cloud SQL part. 

### Steps to reproduce

I don't know if it is easily reproducable, it happened to me without any change on my stack...
It happens on my dev stack, I haven't try yet on production.

Here is my code for the SQL Instance:

```
export const apiDatabaseInstance = new gcp.sql.DatabaseInstance(
  `sql-api-${env}`,
  {
    databaseVersion: 'POSTGRES_13',
    region: gcp.config.region,
    settings: {
      tier: dbAPIMachineType,
      backupConfiguration: {
        enabled: env === 'prod',
        backupRetentionSettings: {
          retainedBackups: env === 'prod' ? 60 : 7,
        },
      },
    },
  },
);

export const apiDatabase = new gcp.sql.Database(`sql-api-${env}`, {
  instance: apiDatabaseInstance.name,
  name: `sql-api-${env}`,
});

export const apiDatabaseUser = new gcp.sql.User(`admin-${env}`, {
  instance: apiDatabaseInstance.name,
  name: `admin`,
  password: getSecret('SQL_DATABASE_PASSWORD'),
});
```

### Expected Behavior

The deployment must succeed.
At least I would like to have an error message and some context on what happened.

### Actual Behavior

When I run the command pulumi up, the deployment is failing.
Pulumi try to update the sql cloud instance, even if in the preview the instance is not tagged to be updated.
So I don't know in the first place what pulumi try to do to the Cloud SQL Instance and why an update is triggered.

There is an error but no error message :
![Capture d’écran 2023-01-11 à 09 25 35](https://user-images.githubusercontent.com/6079231/211757193-b5ea2fb2-8b0a-4351-9d92-ae4ab1002c52.png)

On the GCP side, I can see as well a maintenance operation with an error but no log.
![Capture d’écran 2023-01-11 à 09 13 46](https://user-images.githubusercontent.com/6079231/211757308-d74611d8-2431-4c44-b7d3-126bf52d9da8.png)

### Output of `pulumi about`

CLI          
Version      3.51.0
Go Version   go1.19.4
Go Compiler  gc

Plugins
NAME        VERSION
gcp         6.46.0
kubernetes  3.23.1
nodejs      unknown
random      4.8.2

Host     
OS       darwin
Version  13.1
Arch     arm64

This project is written in nodejs: executable='/Users/denislaboureyras/.volta/bin/node' version='v16.13.1'

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by asimov-layton** [comment]
@DenisLaboureyras We experienced something similar that seems to be related to adding `deletionPolicy` to the Resource. It could be unrelated but we noted this recent change: https://github.com/pulumi/pulumi-gcp/commit/38e440091c9e790a55e5b61bcfeebefadd9dfa57#diff-73e583b82eb4aadf5b5fc8c18f126a6735ee5e6a0eefbe4a8b709bc703265ecdR122 (line 122 of `sdk/nodejs/sql/database.ts`)

We resolved the issue by performing a `refresh` on the Resource. After that `up` was able to apply the change in all environments.
